app/src/test/java/com/github/se/signify/model/common/user/SaveUserToFirestoreTest.kt

```diff
@@ -125,4 +130,76 @@ class SaveUserToFirestoreTest {
     // (Since we cannot easily verify logs, we focus on ensuring no Firestore interaction occurs)
     verify(mockFirestore, never()).collection(any())
   }
+
+  @Test
+  fun saveUserToFireStore_shouldLogErrorWhenEmailIsNull() {
+
+    // Mock static FirebaseAuth, FirebaseFirestore, and Log
+    val logMock: MockedStatic<Log> = mockStatic(Log::class.java)
+    val firebaseAuthMock: MockedStatic<FirebaseAuth> = mockStatic(FirebaseAuth::class.java)
+    val firebaseFirestoreMock: MockedStatic<FirebaseFirestore> =
+        mockStatic(FirebaseFirestore::class.java)
+
+    try {
+      // Arrange
+      `when`(FirebaseAuth.getInstance()).thenReturn(mockAuth)
+      `when`(mockAuth.currentUser).thenReturn(mockCurrentUser)
+      `when`(mockCurrentUser.email).thenReturn(null)
+
+      // Prevent Firestore from initializing
+      val mockFirestore = mock(FirebaseFirestore::class.java)
+      `when`(FirebaseFirestore.getInstance()).thenReturn(mockFirestore)
+
+      // Act
+      saveUserToFirestore()
+
+      // Verify Log.e is called with the correct message
+      logMock.verify(
+          { Log.e(eq("FireStore"), eq("Cannot save user: email is null or blank")) }, times(1))
+
+      // Ensure Firestore is NOT accessed
+      verify(mockFirestore, never()).collection(any())
+    } finally {
+      // Clean up static mocks
+      logMock.close()
+      firebaseAuthMock.close()
+      firebaseFirestoreMock.close()
+    }
+  }
+
+  @Test
+  fun saveUserToFireStore_shouldLogErrorWhenEmailIsBlank() {
+
+    // Mock static FirebaseAuth, FirebaseFirestore, and Log
+    val logMock: MockedStatic<Log> = mockStatic(Log::class.java)
+    val firebaseAuthMock: MockedStatic<FirebaseAuth> = mockStatic(FirebaseAuth::class.java)
+    val firebaseFirestoreMock: MockedStatic<FirebaseFirestore> =
+        mockStatic(FirebaseFirestore::class.java)
+
+    try {
+      // Arrange
+      `when`(FirebaseAuth.getInstance()).thenReturn(mockAuth)
+      `when`(mockAuth.currentUser).thenReturn(mockCurrentUser)
+      `when`(mockCurrentUser.email).thenReturn("   ") // Simulate blank email
+
+      // Prevent Firestore from initializing
+      val mockFirestore = mock(FirebaseFirestore::class.java)
+      `when`(FirebaseFirestore.getInstance()).thenReturn(mockFirestore)
+
+      // Act
+      saveUserToFirestore()
+
+      // Verify Log.e is called with the correct message
+      logMock.verify(
+          { Log.e(eq("FireStore"), eq("Cannot save user: email is null or blank")) }, times(1))
```

Comment:
You are also using the string "Cannot save user: email is null or blank" twice, you can also define it in a constant.